mne_bids/path.py

```diff
@@ -405,6 +405,25 @@ def update(self, check=None, **entities):
         self._check()
         return self
 
+    def match(self):
+        """Get a list of all matching basenames.
+
+        Performs a search based on the entities that are set in the
+        ``BIDSPath`` object. Requires a ``root`` attribute to be set.
+
+        Returns
+        -------
+        paths : List[BIDSPath]
```

Comment:
Did we agree to use typo annotations? I'd vote for 

- first make a decision
- introduce the annotations across the entire code base at once

cc @agramfort @sappelhoff @jasmainak